### ISSUE TITLE
Fedora 21: Bumblebee stopped working with Kernels > 3.17.7

### ISSUE THREAD

**Issue by keckelt** [Bumblebee-Project/Bumblebee#631]
**Distro:** Fedora 21 - Workstatioen (FedUp from Fedora 20)
**Kernel:** 3.17.8-300.fc21.x86_64 and 3.18.3-201.fc21.x86_64
**Manufactorer and model:**
baseboard-manufacturer: CLEVO CO.                       
baseboard-product-name: W35_37ET
baseboard-version     : N/A                             
system-manufacturer   : CLEVO CO.                       
system-product-name   : W35_37ET
system-version        : N/A                             
bios-vendor           : American Megatrends Inc.
bios-version          : 4.6.5
bios-release-date     : 11/02/2012

**Bumblebee Version:** optirun (Bumblebee) 3.2.1

Hi, I discovered that Bumblebee stopped working after updating my Kernel.
The latest version i have installed is 3.18.3-201 and there isn't even the bumblee deamon running (calling **optirun** in terminal), nor can't I start it as modprobe doesn't find the bbswitch module (**bumblebeed --daemon**). I tried to re-install both, bbswitch and bumblebee, via yum but that didn't change anything.
 When i boot 3.17.7-300 from GRUB it still works and my dedicated NVIDIA GPU is turned off; but also **bumblebeed --daemon** displays:
*modprobe: FATAL: Module bbswitch not found.*

I recently installed VirtualBox, following the instructions given here:
http://www.if-not-true-then-false.com/2010/install-virtualbox-with-yum-on-fedora-centos-red-hat-rhel/

Besides that, i didn't do anything more unusual than  *yum update*

Any tips appreciated.

**Comment by karolherbst** [comment]
it works for me with 3.8.0 - 3.8.5, so I guess it may be your setup. "modprobe: FATAL: Module bbswitch not found." indicates, that the required modules are not built for your kernel.

**Comment by keckelt** [comment]
It worked for over a year now (including 3.17.7 kernel) and I (think I) didn't change anything.

When the module is not built for my kernel, how can it still work?

**Comment by karolherbst** [comment]
I think the issue is, that modprobe doesn't find bbswitch anymore for some reasons

**Comment by gsgatlin** [comment]
Does running these commands help? (as root)

dkms add -m bbswitch -v 0.8.0
dkms build -m bbswitch -v 0.8.0
dkms install -m bbswitch -v 0.8.0

**Comment by keckelt** [comment]
Thanks for the answers, I got held up in the afternoon:
Manually, I can get the daemon running with:
```
$ sudo modprobe nouveau 
$ bumblebeed --daemon
```

will try dkms now

**Comment by karolherbst** [comment]
> Has this something to do with the fact that nvidia and nouveau are both blacklisted in /etc/modprobe.d/bumblebee.conf ?

no, this is no problem at all. You should be able to run 'sudo modprobe nvidia' at any time.

**Comment by keckelt** [comment]
Hmm,
After changing */etc/modprobe.d/bumblebee.conf * to:
```
blacklist nvidia
#blacklist nouveau
```
bumblebee works as expected now, i.e. Intel card is used after boot and I can use optirun.

**Comment by gsgatlin** [comment]
See issue #582 for using bumblebee with nouveau driver in fedora. My experiences were that bumblebee-nouveau package was required. (and  xorg-x11-drv-modesetting had to be removed) But it sounds like maybe you've got it figured out now? Were you trying to use nouveau or nvidia module with bumblebee?

**Comment by keckelt** [comment]
I wasn't using the *bumblebee-nouveau*-package so I installed it now and reverted the changes in */etc/modprobe.d/bumblebee.conf*, but the problems are the same as before.
I will stick to the "solution" above and comment out the blacklisting of nouveau.

**Comment by ArchangeGabriel** [comment]
Hum, this could be autodetection not working if nouveau isn’t loaded. Should take a look at that.
